api/v1alpha1/nodepool_types.go

```diff
@@ -38,7 +38,18 @@ type NodePoolSpec struct {
 	NodeCount *int32 `json:"nodeCount"`
 	// +optional
 	AutoScaling *NodePoolAutoScaling `json:"autoScaling,omitempty"`
-	Platform    NodePoolPlatform     `json:"platform"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default={maxSurge: 1, maxUnavailable: 0}
+	Management NodePoolManagement `json:"nodePoolManagement"`
+	Platform   NodePoolPlatform   `json:"platform"`
+
+	// Version is the semantic version of the release applied by
+	// the hosted control plane operator.
+	// For a nodePool a given version dictates the ignition config and
+	// an image artifact e.g an AMI in AWS.
+	// +kubebuilder:validation:Optional
+	Version string `json:"version,omitempty"`
```

Comment:
is this really written by hosted control plane operator, or directly be end-user of mgmt cluster?